app/views/admin/purchases/new.html.erb

```diff
@@ -0,0 +1,25 @@
+<%= form_for @purchase do |f| %>
+  <%= f.label :transaction_date %>
+  <%= f.text_field :transaction_date %>
+
+  <%= f.label :payment_date %>
+  <%= f.text_field :payment_date %>
+
+  <%= f.label :amount %>
+  <%= f.number_field :amount, class: :text_field, step: :any %>
+
+  <%= f.label :description %>
+  <%= f.text_area :description %>
+
+  <%= f.fields_for :program_vendor_purchases do |ff| %>
```

Comment:
I would like help or review on this f.fields_for function. I'm trying to create a purchase, which means creating an entry in my joins table that logs the purchase, the vendor, and the program. I can render the form and select options, but it then fails at line 4 of my purchases controller with the following error:

<img width="1215" alt="screen shot 2018-04-04 at 12 37 51 pm" src="https://user-images.githubusercontent.com/5290130/38327269-0d830fac-3805-11e8-94d0-d567894b9b78.png">

I think this means I'm passing in the program id instead of the program, but I don't know how to change the collection select. Any pointers @iandouglas or @AliSchlereth ?

Reply:
@ssciolist I've not used a fields_for before so any help I could offer would come from googling things. 

Outside of that, here are some thoughts:
* Is a purchase what you ended up calling transactions or is this something different?
* Usually what I would see would be on the purchase form, drop downs for program, vendor, and program. These would send through an id I believe then in your controller you would handle the logic of making sure the ids get associated to the right type of resource.  Does that make sense?

Reply:
Hmm, yes let me know if the form_for allows that. I'm curious.

If that doesn't work, it sounds like you can move forward with this fields_for and use the ids in your controller to create the association.